libs/core/src/lib/utils/directives/truncation/truncation.directive.ts

```diff
@@ -0,0 +1,55 @@
+import { AfterViewInit, Directive, ElementRef, Input } from '@angular/core';
+
+@Directive({
+    // eslint-disable-next-line @angular-eslint/directive-selector
+    selector: '[fd-truncate]'
+})
+export class TruncationDirective implements AfterViewInit {
+    @Input() charLength: number;
+    @Input() pixLength: number;
+
+    /** @hidden */
+    private defaultTruncationByChar = 30;
+
+    constructor(private _elementRef: ElementRef) {}
+
+    /** @hidden */
+    ngAfterViewInit(): void {
```

Comment:
I think we'll want to run this with `ngAfterViewChecked` rather than init, in case the text changes after init